src/App.js

```diff
@@ -1,27 +1,53 @@
-import React, { Suspense } from "react";
+import React, { useState, Suspense, useEffect } from "react";
 import "./App.css";
-import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { ThemeProvider, createTheme } from "@mui/material/styles";
 import Routes from "./routes/Routes";
 import { AuthProvider } from "./utils/Auth/use-auth";
 import { QueryClient, QueryClientProvider } from "react-query";
 import { ReactQueryDevtools } from "react-query/devtools";
-
-const theme = createTheme();
+import ThemeSwitcher from "./components/ThemeSwitcher/themeSwitcher";
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
 
 function App() {
   const queryClient = new QueryClient();
+  const [mode, setMode] = useState(() => {
+    let localMode = localStorage.getItem('mode') || 'light'
```

Comment:
Performance suggestion:
```
let localMode = localStorage.getItem('mode') || 'light'
```
the value of the `localMode` is not changeing, so you should use `const` instead of `let`